### DISCUSSION TITLE
led control?

### DISCUSSION THREAD

**Discussion by EXCISIONMODZ** [mriscoc/Ender3V2S1#212] [Ideas]
can you make a menu for controlling leds on the skr mini e3 v3 board?? please

**Comment by mriscoc** [comment]
It is available if you compile your version following this sponsored configuration: 
https://github.com/mriscoc/Special_Configurations/releases/tag/Neopixel

**Reply by EXCISIONMODZ** [reply]
any chance you can help me compile it

**Reply by EXCISIONMODZ** [reply]
i just dont know what im doing

**Reply by EXCISIONMODZ** [reply]
they are neo grb 114 pixels

**Reply by EXCISIONMODZ** [reply]
> It is available if you compile your version following this sponsored configuration: https://github.com/mriscoc/Special_Configurations/releases/tag/Neopixel

i cant get it to work i keep getting errors

**Reply by mriscoc** [reply]
I sent you a message on Patreon. Please give me the details about your setup for be able to help you.

**Reply by EXCISIONMODZ** [reply]
> I sent you a message on Patreon. Please give me the details about your setup for be able to help you.

i sent the info

**Reply by mriscoc** [reply]
ok, let me have time to check it.

**Reply by EXCISIONMODZ** [reply]
> ok, let me have time to check it.

np thank you very much id love to learn if you ever want to help me learn

**Reply by EXCISIONMODZ** [reply]
> ok, let me have time to check it.

did you happen to check it out for me

**Reply by Nbrooks639** [reply]
did you get the neos working?

**Reply by EXCISIONMODZ** [reply]
> did you get the neos working?

dont ask me

**Reply by EXCISIONMODZ** [reply]
> did you get the neos working?

still waiting

**Reply by Nbrooks639** [reply]
what are u waiting for?

**Reply by EXCISIONMODZ** [reply]
for him to compile it
if you dont need him to compile it then its pretty easy as there is a set of pins for the leds

**Reply by EXCISIONMODZ** [reply]
> what are u waiting for?

i dont know how to compile it or anything as i dont know what im doing and i get errors

**Reply by EXCISIONMODZ** [reply]
this is only here because im having trouble getting everything compiled thats it

**Comment by EXCISIONMODZ** [comment]
any update

**Reply by Nbrooks639** [reply]
yea i am the one who helped him set them up im trying to help you out. what errors are u getting

**Reply by EXCISIONMODZ** [reply]
all kind because idk what im doing

**Reply by EXCISIONMODZ** [reply]
> yea i am the one who helped him set them up im trying to help you out. what errors are u getting

what json is he reffering to

**Reply by EXCISIONMODZ** [reply]
> yea i am the one who helped him set them up im trying to help you out. what errors are u getting

In file included from Marlin\src\HAL\STM32\../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\HAL.cpp:30:
Marlin\src\HAL\STM32\../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
Compiling .pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\tft\gt911.cpp.o
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\MarlinSerial.h:24,
                 from Marlin\src\HAL\STM32\HAL.h:34,
                 from Marlin\src\HAL\STM32\HAL.cpp:27:
Marlin\src\HAL\STM32\../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\../../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\HAL_SPI.cpp:27:
Marlin\src\HAL\STM32\../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\MarlinSerial.cpp:27:
Marlin\src\HAL\STM32\../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\HAL_SPI.cpp:27:
Marlin\src\HAL\STM32\../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\Servo.cpp:27:
Marlin\src\HAL\STM32\../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\MarlinSerial.cpp:27:
Marlin\src\HAL\STM32\../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\eeprom_bl24cxx.cpp:32:
Marlin\src\HAL\STM32\../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\Servo.cpp:27:
Marlin\src\HAL\STM32\../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\eeprom_flash.cpp:27:
Marlin\src\HAL\STM32\../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\eeprom_bl24cxx.cpp:32:
Marlin\src\HAL\STM32\../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\eeprom_if_iic.cpp:32:
Marlin\src\HAL\STM32\../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\eeprom_sram.cpp:27:
Marlin\src\HAL\STM32\../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\eeprom_flash.cpp:27:
Marlin\src\HAL\STM32\../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\eeprom_sdcard.cpp:31:
Marlin\src\HAL\STM32\../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\eeprom_if_iic.cpp:32:
Marlin\src\HAL\STM32\../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\eeprom_wired.cpp:27:
Marlin\src\HAL\STM32\../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\fast_pwm.cpp:27:
Marlin\src\HAL\STM32\../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\eeprom_sram.cpp:27:
Marlin\src\HAL\STM32\../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\eeprom_sdcard.cpp:31:
Marlin\src\HAL\STM32\../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\eeprom_wired.cpp:27:
Marlin\src\HAL\STM32\../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\fastio.cpp:27:
Marlin\src\HAL\STM32\../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\fast_pwm.cpp:27:
Marlin\src\HAL\STM32\../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\fastio.cpp:27:
Marlin\src\HAL\STM32\../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
Compiling .pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\tft\tft_fsmc.cpp.o
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\HAL.cpp.o] Error 1
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\MarlinSerial.cpp.o] Error 1
In file included from Marlin\src\HAL\STM32\../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\sdio.cpp:27:
Marlin\src\HAL\STM32\../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\Servo.cpp.o] Error 1
In file included from Marlin\src\HAL\STM32\tft\../../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\tft\../../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\tft\gt911.cpp:26:
Marlin\src\HAL\STM32\tft\../../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\HAL_SPI.cpp.o] Error 1
In file included from Marlin\src\HAL\STM32\../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\sdio.cpp:27:
Marlin\src\HAL\STM32\../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
In file included from Marlin\src\HAL\STM32\tft\../../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\tft\../../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\tft\gt911.cpp:26:
Marlin\src\HAL\STM32\tft\../../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\tft\../../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\tft\../../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\eeprom_bl24cxx.cpp.o] Error 1
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\eeprom_flash.cpp.o] Error 1
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\eeprom_if_iic.cpp.o] Error 1
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\eeprom_sram.cpp.o] Error 1
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\eeprom_sdcard.cpp.o] Error 1
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\eeprom_wired.cpp.o] Error 1
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\fast_pwm.cpp.o] Error 1
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\fastio.cpp.o] Error 1
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\sdio.cpp.o] Error 1
In file included from Marlin\src\HAL\STM32\tft\../../../inc/../pins/pins.h:541,
                 from Marlin\src\HAL\STM32\tft\../../../inc/MarlinConfig.h:35,
                 from Marlin\src\HAL\STM32\tft\tft_fsmc.cpp:27:
Marlin\src\HAL\STM32\tft\../../../inc/../pins/stm32g0/pins_BTT_SKR_MINI_E3_V3_0.h:173:4: warning: #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue." [-Wcpp]
  173 |   #warning "DWIN_CREALITY_LCD requires a custom cable, see diagram above this line. Comment out this line to continue."
      |    ^~~~~~~
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\tft\gt911.cpp.o] Error 1
In file included from Marlin\src\HAL\STM32\tft\../../../inc/MarlinConfigPre.h:39,
                 from Marlin\src\HAL\STM32\tft\../../../inc/MarlinConfig.h:28,
                 from Marlin\src\HAL\STM32\tft\tft_fsmc.cpp:27:
Marlin\src\HAL\STM32\tft\../../../inc/../../Configuration.h:3048:42: error: missing binary operator before token "4"
 3048 |   #define NEOPIXEL_PIN     PA8           4 // LED driving pin
      |                                          ^
Marlin\src\HAL\STM32\tft\../../../inc/../core/macros.h:275:51: note: in expansion of macro 'NEOPIXEL_PIN'
  275 | #define PIN_EXISTS(PN)      (defined(PN##_PIN) && PN##_PIN >= 0)
      |                                                   ^~
Marlin\src\HAL\STM32\tft\../../../inc/SanityCheck.h:2718:8: note: in expansion of macro 'PIN_EXISTS'
 2718 |   #if !PIN_EXISTS(NEOPIXEL) || NEOPIXEL_PIXELS == 0
      |        ^~~~~~~~~~
*** [.pio\build\STM32G0B1RE_btt\src\src\HAL\STM32\tft\tft_fsmc.cpp.o] Error 1
========================================================== [FAILED] Took 25.01 seconds ==========================================================

Environment      Status    Duration
---------------  --------  ------------
STM32G0B1RE_btt  FAILED    00:00:25.010
===================================================== 1 failed, 0 succeeded in 00:00:25.010 =====================================================

**Reply by EXCISIONMODZ** [reply]
https://prnt.sc/6DaWrcHtc3mT

**Reply by mriscoc** [reply]
> https://prnt.sc/6DaWrcHtc3mT

Remove that "4"

**Comment by mriscoc** [comment]
OK based upon your replies your json configuration file for the Neopixel is:
```json
{
"Configuration_adv.h" : [
  {
    "op": "Enable",
    "searchfor": "LED_CONTROL_MENU",
    "comment" : "For use with NeoPixel"
  }
],
"Configuration.h" : [
  {
    "op": "Enable",
    "searchfor": "NEOPIXEL_LED"
  },
  {
    "op": "Custom",
    "searchfor": "NEOPIXEL_TYPE",
    "mask" : "[^ ]*",
    "value": "NEO_GRB"
  },
  {
    "op": "Custom",
    "searchfor": "NEOPIXEL_PIN",
    "mask" : "[^ ]*",
    "value": "PA8"
  },
  {
    "op": "CustomVal",
    "searchfor": "NEOPIXEL_PIXELS",
    "value": "114"
  },
  {
    "op": "Enable",
    "searchfor": "NEOPIXEL_STARTUP_TEST"
  }
],
"Version.h" : [
]   
}
```
The SKR board needs a `screen adapter` for use of the DWIN/DACAI color display of your Ender 3V2. For 300°C you need special thermistor and heat break. The next step is to set the correct thermistor type (`TEMP_SENSOR_0`) in the configuration file, the stock type 1 does not allow temperatures above the 285 °C

For example, for Volcano hotend (type 5), the json configuration file is:

```json
{
"Configuration_adv.h" : [
],
"Configuration.h" : [
  {
    "op": "CustomVal",
    "searchfor": "TEMP_SENSOR_0",
    "value": "5",
    "comment": "Volcano thermistor"
  },
  {
    "op": "CustomVal",
    "searchfor": "HEATER_0_MINTEMP",
    "value": "5",
    "comment": "Volcano thermistor"
  },
  {
    "op": "CustomVal",
    "searchfor": "HEATER_0_MAXTEMP",
    "value": "300",
    "comment": "Volcano thermistor"
  }
],
"Version.h" : [
]   
}
```

**Reply by EXCISIONMODZ** [reply]
what json file

**Reply by mriscoc** [reply]
What thermistor have you? `TEMP_SENSOR_0`

**Reply by EXCISIONMODZ** [reply]
i got all that fixed but i cant get octoprint to work

**Reply by mriscoc** [reply]
Set baud to 250000
https://github.com/mriscoc/Ender3V2S1/wiki/Octoprint

**Comment by mriscoc** [comment]
Were you able to have a successful compile?

**Reply by EXCISIONMODZ** [reply]
yes but the printer will stop responding and acts like its off

**Comment by EXCISIONMODZ** [comment]
![1657401368939613756730486287136](https://user-images.githubusercontent.com/53412982/178122908-c73926e9-6c93-470f-bea8-f3aa0881aba8.jpg)

**Reply by EXCISIONMODZ** [reply]
This what I'm meaning by acting like it's off

**Reply by mriscoc** [reply]
The printer needs a correct adapter to connect the screen unit to the SKR board.
See diagram above this line: pins_BTT_SKR_MINI_E3_V3_0.h:173

**Reply by EXCISIONMODZ** [reply]
it is done correctly 
the entire printer just stops mid print and does this

**Reply by EXCISIONMODZ** [reply]
???????????????????????????

**Reply by mriscoc** [reply]
Sorry but you must have a problem not reported before and it is not caused by the current official firmware. Please review your changes and apply them one by one.
